Pokemon App/Pokemon App/Data Model/PokemonCard.swift

```diff
@@ -11,10 +11,29 @@ import UIKit
 class PokemonCard {
     let id: String
     let imageURL: String
+    var image: UIImage?
     
-    init(id: String, imageURL: String) {
+    init(id: String, imageURL: String, image: UIImage?) {
         self.id = id
-        
         self.imageURL = imageURL
+        self.image = image
+    }
+    
+    func downloadImage() {
+        let loader = ImageLoader()
+        if let url = URL(string: imageURL) {
+            loader.downloadImageWithURLSession(url: url) { result in
+                switch result {
+                case .success(let downloadedImage):
+                    self.image = downloadedImage
+                case .failure:
+                    self.image = nil
+                }
+            }
+        }
+    }
+    
+    func cancelDownloadImage(){
+        image = nil
```

Comment:
Nah ini udah ada fungsinya. tapi kyknya harusnya  lu cuma cancel loadernya aja disini, imagenya gak perlu di nil. nil-image-nya pas dithreshold yg tadi gue jelasin dicomment sebelumnya aja.
```suggestion

        let loader = ImageLoader()
    func downloadImage() {
        if let url = URL(string: imageURL) {
            loader.downloadImageWithURLSession(url: url) { result in
                switch result {
                case .success(let downloadedImage):
                    self.image = downloadedImage
                case .failure:
                    self.image = nil
                }
            }
        }
    }
    
    func cancelDownloadImage(){
        loader.cancelDownloadImage()
```